apps/custom-ui-app-docusign/public/js/panel.js

```diff
@@ -0,0 +1,62 @@
+// Shows a Snack Bar with a message
+async function showSnackBar(message) {
+    const sdk = await new AppExtensionsSdk().initialize();
+    await sdk.execute(AppExtensionsSdk.Command.SHOW_SNACKBAR, {
+        message
+    });
+}
+
+// Shows a modal with a given data
+async function showModal(data) {
+    const sdk = await new AppExtensionsSdk().initialize();
+    await sdk.execute(AppExtensionsSdk.Command.OPEN_MODAL, {
+        type: AppExtensionsSdk.Modal.JSON_MODAL,
+        action_id: '5c5fb865-f967-4bc2-88eb-96cae0ee74e7', // ID associated with the Custom UI Modal (visible in Marketplace Manager)
+        data
+    });
+};
+
+// Sends the document to the contact associated with the deal
+$(document).on('click', '.send-doc', function (e) {
+    const template_id = e.target.dataset.id;
+    $.ajax({
+        type: "POST",
+        url: "/api/send_document",
+        data: JSON.stringify({
+            template_id,
+            deal_id,
+            company_id
+        }),
+        contentType: "application/json; charset=utf-8",
+        dataType: "json",
+        success: function (data) {
+            showSnackBar('Document sent successfully');
+        },
+        error: function (errMsg) {
+            showSnackBar('Sending document failed. Check the console for details.');
+            console.error(errMsg);
+        }
+    });
+});
+
+// Generates a preview by requesting the preview session URL
+$(document).on('click', '.preview-doc', function (e) {
+    const template_id = e.target.dataset.id;
+    $.ajax({
```

Comment:
Can you use Fetch instead of ajax? https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API it's built into the browser